test/interop.jl

```diff
@@ -194,23 +194,42 @@ end
         # FIXME: Win32 nightly emits a i64*, even though bitstype_to_llvm uses T_int8
         @test_broken contains(ir, r"@julia_unsafe_load_\d+\(i8\*")
     else
-        @test contains(ir, r"@julia_unsafe_load_\d+\(i8\*")
+        if LLVM.version() < v"15"
+            @test contains(ir, r"@julia_unsafe_load_\d+\(i8\*")
+        else
+            @test contains(ir, r"@julia_unsafe_load_\d+\(ptr")
+        end
+    end
+    if LLVM.version() < v"15"
+        @test contains(ir, r"load i64, i64\* %\d+, align 1")
+    else
+        @test contains(ir, r"load i64, ptr %\d+, align 1")
     end
-    @test contains(ir, r"load i64, i64\* %\d+, align 1")
-
     ir = sprint(io->code_llvm(io, unsafe_load, Tuple{typeof(ptr), Int, Val{4}}))
-    @test contains(ir, r"load i64, i64\* %\d+, align 4")
+    if LLVM.version() < v"15"
+        @test contains(ir, r"load i64, i64\* %\d+, align 4")
+    else
+        @test contains(ir, r"load i64, ptr %\d+, align 4")
+    end
 end
 
 @testset "reinterpret with addrspacecast" begin
     ptr = reinterpret(Core.LLVMPtr{Int64, 4}, 0)
     for eltype_dest in (Int64, Int32), AS_dest in (4, 3)
         T_dest = Core.LLVMPtr{eltype_dest, AS_dest}
         ir = sprint(io->code_llvm(io, LLVM.Interop.addrspacecast, Tuple{Type{T_dest}, typeof(ptr)}))
-        if AS_dest == 3
-            @test contains(ir, r"addrspacecast i8 addrspace\(4\)\* %\d+ to i8 addrspace\(3\)\*")
+        if LLVM.version() < v"15"
```

Comment:
`LLVM.supports_typed_pointers(ctx)`?

Reply:
There isn't a ctx here currently. I guess we could add one?

Reply:
As `supports_typed_pointers` isn't going to change between contexts, maybe we should determine it at the start of `runtests.jl` and just use that global?

Reply:
On the tests themselves, since we're testing the default value it is version based, but I guess that can work.